Makefile

```diff
@@ -189,6 +189,7 @@ init-coverage: ## Initialize converage report for Cilium unit-tests.
 	$(QUIET) echo "mode: count" > coverage-all-tmp.out
 	$(QUIET) echo "mode: count" > coverage.out
 
+unit-tests: GO_TAGS_FLAGS+=integration_tests
```

Comment:
I agree that some of these tests are integration tests, and tagging them as such is a good step forward.

What I like about this: Calling things out explicitly.

What I don't like about this particular line: Now we're explicitly defining the target `unit-tests` as integration tests, even though the majority of the tests are actually unit tests.

I'd like to see these to be part of a separate make target if we're going to call them out separately - maybe something like `make integration-tests` or `make go-integration-tests`?

At the same time, I don't want to lose coverage over these so if these particular tests are being run via a different target then let's make sure to grep around for where the `unit-tests` target is used, and update those locations to also call this target.

If we split this out, we can also move the `start-kvstores` / `stop-kvstores` logic over there, which will speed up this target. We can also update the docs so that we're not telling developers "to run the integration tests, run `make unit-tests`."

Reply:
OK, I've renamed the `unit-tests` target to `integration-tests` and updated the documentation, which should resolve this.